ml-agents/mlagents/trainers/tests/test_stats.py

```diff
@@ -0,0 +1,76 @@
+import unittest.mock as mock
+import os
+
+from mlagents.trainers.stats import StatsReporter, TensorboardWriter
+
+
+def test_stat_reporter_add_summary_write():
+    # Test add_writer
+    StatsReporter.writers.clear()
+    mock_writer1 = mock.Mock()
+    mock_writer2 = mock.Mock()
+    StatsReporter.add_writer(mock_writer1)
+    StatsReporter.add_writer(mock_writer2)
+    assert len(StatsReporter.writers) == 2
+
+    # Test add_stats and summaries
+    statsreporter1 = StatsReporter("category1")
+    statsreporter2 = StatsReporter("category2")
+    for i in range(10):
+        statsreporter1.add_stat("key1", float(i))
+        statsreporter2.add_stat("key2", float(i))
+
+    statssummary1 = statsreporter1.get_stats_summaries("key1")
+    statssummary2 = statsreporter2.get_stats_summaries("key2")
+
+    assert statssummary1.num == 10
+    assert statssummary2.num == 10
+    assert statssummary1.mean == 4.5
+    assert statssummary2.mean == 4.5
+    assert round(statssummary1.std, 1) == 2.9
```

Comment:
I think you can also do `assert statssummary1.std == pytest.approx(2.9)` - maybe a little cleaner.

Reply:
`pytest.approx(2.9, abs=0.1)` does the same thing I think